api/app/Http/Controllers/MediaController.php

```diff
@@ -186,82 +217,282 @@ public function showOneMedia($id){
+	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="caption",
+	 *		 in="query",
+	 *		 description="Consist, free-form text describing content of media object",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=100)
+	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="copyright",
+	 *		 in="query",
+	 *		 description="Copyright Owner",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=255)
+	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="rights",
+	 *		 in="query",
+	 *		 description="A URI pointing to structured information about rights held in and over the resource (e.g. https://creativecommons.org/publicdomain/zero/1.0/, https://creativecommons.org/licenses/by/4.0/legalcode)",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=255)
+	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="accessRights",
+	 *		 in="query",
+	 *		 description="Information about who can access the resource or an indication of its security status",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=255)
+	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="locality",
+	 *		 in="query",
+	 *		 description="Short description of locality. Leave null if locality is available via the occurrence record",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=250)
+	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="notes",
+	 *		 in="query",
+	 *		 description="General notes",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=350)
+	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="sourceIdentifier",
+	 *		 in="query",
+	 *		 description="The source identifier, if media was harvested from an external resource",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=150)
+	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="hashFunction",
+	 *		 in="query",
+	 *		 description="Cryptographic hash function used to compute the value given in the Hash Value",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=45)
+	 *	 ),
+	 *	 @OA\Parameter(
+	 *		 name="hashValue",
+	 *		 in="query",
+	 *		 description="Value computed by a hash function applied to the media that will be delivered at the access point",
+	 *		 required=false,
+	 *		 @OA\Schema(type="string", maxLength=45)
+	 *	 ),
+	 *	 @OA\Response(
+	 *		 response="201",
+	 *		 description="Returns UUID recordID (GUID)",
+	 *		 @OA\JsonContent()
+	 *	 ),
+	 *	 @OA\Response(
+	 *		 response="400",
+	 *		 description="Error: Bad request.",
+	 *	 ),
+	 *	 @OA\Response(
+	 *		 response="401",
+	 *		 description="Unauthorized",
+	 *	 ),
+	 * )
+	 */
 	public function update($id, Request $request){
-		$media = Media::findOrFail($id);
-		$media->update($request->all());
+		if($user = $this->authenticate($request)){
+			$media = Media::findOrFail($id);
+			$this->validate($request, $this->rulesUpdate);
 
-		return response()->json($media, 200);
+			$inputArr = $request->all();
+			$this->adjustInputData($inputArr);
+			$media->update($inputArr);
+			return response()->json($media, 200);
```

Comment:
Again, I'm not sure whether a 200 response gets returned even if some of these calls upstream don't go as expected.